src/realm/object-store/property.hpp

```diff
@@ -390,24 +391,40 @@ inline size_t Property::type_nesting_levels() const noexcept
 
 inline std::string Property::type_string() const
 {
+    std::stringstream nested;
```

Comment:
Probably a bit overkill to use stringstream here, You are just concatenating constant strings.